src/cca/app/scripts/cordova_browser_api.ts

```diff
@@ -50,14 +50,43 @@ class CordovaBrowserApi implements BrowserAPI {
   }
 
   public startUsingProxy = (endpoint:net.Endpoint) => {
-    // TODO: Implement getter support, possibly using "redsocks".
+    if (!chrome.proxy) {
+      console.log('No proxy setting support; ignoring start command');
+      return;
+    }
+
+    chrome.proxy.settings.set({
+      scope: "regular",
+      value: {
+        mode: "fixed_servers",
+        rules: {
+          singleProxy: {
+            scheme: "socks5",
+            host: endpoint.address,
+            port: '' + endpoint.port
+          }
+        }
+      }
+    }, (response:Object) => {
+      console.log('Set proxy response:', response);
+      // Open the in-app browser through the proxy.
+      cordova.InAppBrowser.open('https://www.uproxy.org/', '_blank');
+    });
   };
 
   public stopUsingProxy = () => {
+    if (!chrome.proxy) {
+      console.log('No proxy setting support; ignoring stop command');
+      return;
+    }
+
+    chrome.proxy.settings.clear({scope: "regular"}, () => {
+      console.log('Cleared proxy settings');
+    });
   };
 
   public openTab = (url :string) => {
-    // TODO: Figure out what this means in Cordova.
+    cordova.InAppBrowser.open(url, '_blank');
```

Comment:
Should you open a tab if the proxy hasn't been set? Perhaps you could include a check for `chrome.proxy`, as above for `startUsingProxy` and `stopUsingProxy`.